To get groups claim for Group Overage scenario for Web API

Shama-K opened the pull request:
Updated Web API code to get all the groups in case of groups overage.

Shama-K committed:
To get groups claim for Group Overage scenario for Web API

derisen reviewed (commented):
Works like a charm, thanks Shama! Could you also update the readme as well, adding instructions about 'client secret' and 'offline_access' scopes to the api registration part? Also updating the 'About the Code' section / api part would be great.

Shama-K committed:
Updated comments in the code and Readme

derisen reviewed (approved):
LGTM!

Shama-K committed:
Updated groups overage method

derisen reviewed (approved):
Just installed and tested again. LGTM!

Shama-K committed:
Changes related to MIW 3.0

Shama-K committed:
Merged master branch changes

derisen committed:
updating scripts for implicit flow check

Shama-K committed:
Config changes

idg-sam commented:
Looks good to me!
1. Should we grant admin consent in the instructions?
2. We should change existing manual steps for groups emission to tell user exactly how to add it in IDT and AT
3. New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
4. 3. New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

idg-sam reviewed (changes requested):
Looks good to me!

- Should grant admin consent in the instructions
- In script-based configuration:
   - ~~We should change additional manual stepsfor groups emission to tell user exactly how to add it in IDT and AT~~ Remove the manual step for group claim emission in IDT and AT, since it is already in place 
   - New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
   - New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

TiagoBrenck reviewed (approved):
I am approving, but with the same observations as Sam

derisen committed:
update msal

Shama-K committed:
Minor Updates

Shama-K committed:
Merge branch 'shama-k/WebAPI-GroupOverage' of https://github.com/Azure-Samples/ms-identity-javascript-angular-spa-dotnetcore-webapi-roles-groups into shama-k/WebAPI-GroupOverage

Shama-K commented:
> Looks good to me!
> 
> * Should grant admin consent in the instructions
> * In script-based configuration:
>   
>   * ~We should change additional manual stepsfor groups emission to tell user exactly how to add it in IDT and AT~ Remove the manual step for group claim emission in IDT and AT, since it is already in place
>   * New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
>   * New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

Addressed the suggestions.

Shama-K committed:
Code Refactoring

Shama-K committed:
Readme Updates and Minor Code Changes

Shama-K committed:
Check Scope claim type

kalyankrishna1 reviewed (approved):
:shipit:

kalyankrishna1 commented on chapter2/README.md:
````diff
@@ -395,7 +398,113 @@ ngOnInit(): void {
+    **ProcessClaimsForGroupsOverage** method uses `GraphServiceClient` to retrieve groups for the signed-in user from [/me/memberOf](https://docs.microsoft.com/graph/api/user-list-memberof) endpoint. All the group ids are stored in Session.
+
+    ```csharp
+                private static async Task ProcessUserGroupsForOverage(TokenValidatedContext context)
````
>ProcessUserGroupsForOverage [](start = 42, length = 27)

you dont have to print the whole method

Shama-K replied:
Updated the Readme files.
Thanks.

Shama-K committed:
Removed ProcessUserGroupsForOverage From Readme

derisen reviewed (approved):
Great work Shama, thanks. Just make sure to pull in the latest changes to master before merging.

Shama-K committed:
Merge branch 'master' into shama-k/WebAPI-GroupOverage
